localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_map/state_map.py

```diff
@@ -176,8 +179,13 @@ def _eval_execution(self, env: Environment) -> None:
         frame.stack = copy.deepcopy(env.stack)
 
         try:
-            if self.items_path:
-                self.items_path.eval(env=env)
+            # ItemsPath in DistributedMap states is only used if a JSONinput is passed from the previous state.
+            if (
+                not isinstance(self.iteration_component, DistributedIterationComponent)
```

Comment:
sidenote: complexity is with instance check is probably something to keep an eye on in the future if this would become a common pattern, but it's good to learn about such special cases first :)